src/Components/ArtsyShippingEstimate.tsx

```diff
@@ -0,0 +1,348 @@
+import type ArtaEstimate from "@artaio/arta-browser/dist/estimate"
+import { Link, Spacer, Text } from "@artsy/palette"
+import { useLoadScript } from "Utils/Hooks/useLoadScript"
+import { getENV } from "Utils/getENV"
+import type { ArtsyShippingEstimate_artwork$key } from "__generated__/ArtsyShippingEstimate_artwork.graphql"
+import { useEffect, useState } from "react"
+import { graphql, useFragment } from "react-relay"
+import styled from "styled-components"
+
+const ARTA_API_KEY = getENV("ARTA_API_KEY")
+
+const WIDGET_TITLE = "Estimate Shipping Cost"
+
+type Arta = typeof import("@artaio/arta-browser").default
+
+type EstimateWidgetState = {
+  loaded: boolean
+  widget: ArtaEstimate | null
+}
+
+interface Props {
+  artwork: ArtsyShippingEstimate_artwork$key
+}
+
+const Loader = () => <Spacer y={2} />
+
+const LinkButton = styled(Link)`
+  color: ${props => props.theme.colors.black60};
+  &:hover {
+    text-decoration: underline;
+    color: ${props => props.theme.colors.black100};
+  }
+`
+
+export const ArtsyShippingEstimate = ({ artwork }: Props) => {
+  const artworkData = useFragment(ARTWORK_FRAGMENT, artwork)
+
+  const estimateInput = estimateRequestBodyForArtwork(
+    artworkData as ShippableArtwork,
+  )
+
+  const [Arta, setArta] = useState<Arta | null>(null)
+
+  useLoadScript({
+    id: "arta-browser-script",
+    src: "https://cdn.jsdelivr.net/npm/@artaio/arta-browser@2.16.1/dist/bundle.js",
+    onReady: () => {
+      setArta(window["Arta"])
+    },
+  })
+
+  const initialState: EstimateWidgetState = !!estimateInput
+    ? { loaded: false, widget: null }
+    : { loaded: true, widget: null }
+
+  const [state, setState] = useState<EstimateWidgetState>(initialState)
+
+  // close on unmount
+  useEffect(() => {
+    return () => {
+      if (state.widget?.isOpen) {
+        state.widget.close()
+      }
+    }
+  }, [state.widget])
+
+  useEffect(() => {
+    if (!estimateInput || !ARTA_API_KEY) {
+      setState({ loaded: true, widget: null })
+      return
+    }
+
+    if (state.loaded || !Arta) {
+      return
+    }
+
+    const loadEstimate = async () => {
+      console.log("***", "oading estimate", estimateInput)
+      Arta.init(ARTA_API_KEY)
+
+      const artsyEstimateWidget =
+        Arta &&
+        estimateInput &&
+        Arta.estimate(estimateInput, {
+          text: { header: { title: WIDGET_TITLE } },
+        })
+
+      if (!artsyEstimateWidget) {
+        setState({ loaded: true, widget: null })
+        return
+      }
+
+      try {
+        await artsyEstimateWidget.validate()
+        setState({ loaded: true, widget: artsyEstimateWidget })
+        return
+      } catch (e) {
+        console.error("*** Error validating artsy shipping estimate:", e)
+        setState({ loaded: true, widget: null })
+      }
+    }
+    loadEstimate()
+  }, [Arta, estimateInput, state.loaded, state])
+
+  if (!state.loaded) {
+    return <Loader />
+  }
+
+  if (!state.widget) {
+    return null
```

Comment:
Maybe also return loader here?

Reply:
If it is already loaded and there is no widget, then something failed and/or the artwork isn't able to do a shipping estimate. So in this case we fall back to nothing. 

... though technically the loader is now just a spacer instead of skeleton text so we could keep that to avoid a content jump.